src/tasks/OneDriveLargeFileUploadTask.ts

```diff
@@ -59,7 +59,7 @@ export class OneDriveLargeFileUploadTask extends LargeFileUploadTask {
 		if (path[path.length - 1] !== "/") {
 			path = `${path}/`;
 		}
-		return encodeURI(`/me/drive/root:${path}${fileName}:/createUploadSession`);
+		return `/me/drive/root:${path.split('/').map(p => encodeURIComponent(p)).join('/')}${encodeURIComponent(fileName)}:/createUploadSession`;
```

Comment:
would be nice to have a comment here in case someone attempts to revert it back to the original for optimization.
```suggestion
		// we choose to encode each component of the file path separately because when encoding full URI
		// with encodeURI, special characters like # in the file name doesn't get encoded as desired
		return `/me/drive/root:${path.split('/').map(p => encodeURIComponent(p)).join('/')}${encodeURIComponent(fileName)}:/createUploadSession`;
```